routers/user.controller.js

```diff
@@ -0,0 +1,245 @@
+//         text: `http://localhost:${config.port}/verify/${verificationToken}`,
+//         html: `<p>Please, <a href=http://localhost:${config.port}/auth/verify/${verificationToken}>click</a> to verify your email</p>`,
+//       };
+
+//       sgMail.send(msg);
+//     } catch (err) {
+//       console.log(err);
+//     }
+//   };
+
+//   static validateUserObject = async (req, res, next) => {
+//     try {
+//       const schema = Joi.object({
+//         name: Joi.string(),
+//         email: Joi.string().required(),
+//         password: Joi.string(),
+//       });
+
+//       const validateUser = schema.validate(req.body);
+
+//       if (validateUser.error) {
+//         return res.status(400).send(validateUser.error.details[0].message);
+//       }
+
+//       next();
+//     } catch (err) {
+//       next(err);
+//     }
+//   };
+
+//   static userLogin = async (req, res, next) => {
+//     try {
+//       const { email, password } = req.body;
+
+//       const user = await userModel.findOne({ email });
+
+//       if (!user) {
+//         return res.status(401).send({ message: "Email or password is wrong" });
+//       }
+
+//       console.log("password", password);
+//       console.log("user.passwordHash", user.passwordHash);
+
+//       const isPassValid = await bcrypt.compare(password, user.passwordHash);
+
+//       if (isPassValid) {
+//         return res.status(402).send({ message: "Email or password is wrong" });
+//       }
+
+//       const token = await user.generateAndSaveToken();
+
+//       await userModel.findByIdAndUpdate(user._id, {
+//         token,
+//       });
+
+//       res.status(200).send({
+//         user: {
+//           id: user._id,
+//           name: user.username,
+//           email: user.email,
+//           familyId: user.familyId
+//         },
+//         activeToken: token,
+//       });
+//     } catch (err) {
+//       next(err);
+//     }
+//   };
+// }
+
+// module.exports = userController;
+const Joi = require("joi");
+const bcrypt = require("bcrypt");
+const UserModel = require("../database/modules/user.model");
+const {
+  ApiError,
+  errorHandler,
+  validate,
+  createVerificationToken,
+  mailer,
+} = require("../helpers");
+const authorization = require("../middlewares/auth");
+
+// const authRouter = Router();
+class userController {
+  constructor() {}
+  async userRegister(req, res) {
+    try {
+      validate(
+        Joi.object({
+          name: Joi.string(),
+          email: Joi.string().email().required(),
+          password: Joi.string().min(4),
+        }),
+        req.body
+      );
+
+      const { name, email, password } = req.body;
+
+      const [user] = await UserModel.find({ email });
+
+      if (user) throw new ApiError(409, "Email are in use");
+
+      console.log(typeof createVerificationToken);
+
+      const verificationToken = createVerificationToken();
+
+      await UserModel.create({
+        name,
+        email,
+        passwordHash: password,
```

Comment:
please, use here your hash generation logic and remove it from `pre('save', ...)` hook